StatBot/EchoBot.cs

```diff
@@ -23,49 +36,206 @@ public class EchoBot : IBot
         {
             if (context.Activity.Type == ActivityTypes.Message)
             {
-                if (!String.IsNullOrEmpty(context.Activity.Text))
+                var state = ConversationState<Dictionary<string, object>>.Get(context);
+                var dc = dialogs.CreateContext(context, state);
+                await dc.Continue();
+
+                if (!context.Responded)
                 {
-                    var user = DataModel.RememberUser(context.Activity);
-                    List<ITool> _tools;
-                    _tools = new List<ITool>();
-
-                    var baseInterfaceType = typeof(ITool);
-                    var botCommands = Assembly.GetAssembly(baseInterfaceType)
-                        .GetTypes()
-                        .Where(types => types.IsClass && !types.IsAbstract && types.GetInterface("ITool") != null);
-                    foreach (var botCommand in botCommands)
+                    if (context.Activity.Text.ToLowerInvariant().Contains("/createstat"))
                     {
-                        _tools.Add((ITool)Activator.CreateInstance(botCommand));
+                        await dc.Begin("createStatDialig");
+                        return;
                     }
+                }
 
-                    var str = context.Activity.Text.Trim();
-                    var indexOfSpace = str.IndexOf(" ", StringComparison.Ordinal);
-                    var command = indexOfSpace != -1 ? str.Substring(0, indexOfSpace).ToLower() : str.ToLower();
-                    if (command[0] != '/')
+                if (!String.IsNullOrEmpty(context.Activity.Text))
+                {
+                    var data = context.Activity.Text.Split(' ');
+                    int StatisticId = 0;
+                    string ParrentButtonType = "";
+                    string ButtonType = "";
+                    try
                     {
-                        command = "/" + command;
+                        StatisticId = Convert.ToInt32(data[1]);
+                        ButtonType = data[0];
+                        ParrentButtonType = data[2];
                     }
+                    catch { }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://172.21.13.172:9091/coding_rules#rule_key=csharpsquid%3AS108)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Handle the exception or explain in a comment why it can be ignored. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://172.21.13.172:9091/coding_rules#rule_key=csharpsquid%3AS2486)